cmd/autoscaler/main.go

```diff
@@ -122,7 +124,7 @@ func main() {
 		metrics.ConfigMapWatcher(ctx, component, nil /* SecretFetcher */, logger),
 		profilingHandler.UpdateFromConfigMap)
 
-	podLister := podinformer.Get(ctx).Lister()
+	podLister := filteredpodinformer.Get(ctx, serving.ServiceLabelKey).Lister()
```

Comment:
We shouldn't rely on the ServiceLabelKey here, not all pods might be created from Services (it's totally fine to create them from Configuration or Revision, where that label would then be absent).

Reply:
@markusthoemmes 
How to create pods from Configuration or Revision? And what are these pods used for?

Reply:
@markusthoemmes ,   can you help to explain further what is the use case to create a pod through Configuration or Revision without ksvc CR?    I can understand that a pod can be created with a raw configuration CR directly, but there won't be any route assigned.  Should autoscaler monitor that kinds of pod as well?

Reply:
Yes, Routes have no connection to the autoscaler whatsoever. You are also free to create Configurations/Revisions and create routes manually that point to them.

We could instead use the RevisionUID label for example. That should cover all use-cases (for now)